client/my-sites/controller.js

```diff
@@ -106,19 +105,17 @@ function removeSidebar( context ) {
 			secondary: false,
 		} )
 	);
-	ReactDom.unmountComponentAtNode( document.getElementById( 'secondary' ) );
 }
 
 function renderEmptySites( context ) {
 	const NoSitesMessage = require( 'components/empty-content/no-sites-message' );
 
 	removeSidebar( context );
 
-	renderWithReduxStore(
-		React.createElement( NoSitesMessage ),
-		document.getElementById( 'primary' ),
-		context.store
-	);
+	context.primary = React.createElement( NoSitesMessage );
+
+	makeLayout( context, noop );
+	clientRender( context );
```

Comment:
Just for the record; solving rendering these "no site" warnings by having `makeLayout` and `clientRender` middlewares directly here is not ideal, but given how much refactoring any other method would require I think this is sufficient for the scope of this PR.

Just something we might want to look into again.

Reply:
Maybe add a `TODO` or `FIXME` comment to the code?

Reply:
@ockham good idea. It's not very small todo so I could open an issue as well. Something I prefer anyway over `TODO`-tagging. Do you typically link to GH issues from `TODO` notes in code?

Reply:
Yep, we sometimes do. Good idea in this case IMO 👍